desc/objectives/linear_objectives.py

```diff
@@ -236,6 +236,170 @@ def update_target(self, thing):
+        Should be a scalar or have the same tree structure as thing.params.
+        Unused by this objective (as target is always just zero, the diff btwn the two
+        things' params)
+    normalize : bool, optional
+        Whether to compute the error in physical units or non-dimensionalize.
+        Unused by this objective (as target is always just zero, the diff btwn the two
+        things' params)
+    normalize_target : bool, optional
+        Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True. Has no effect for this objective.
+        Unused by this objective (as target is always just zero, the diff btwn the two
+        things' params)
+    name : str, optional
+        Name of the objective function.
+
+
+    """
+
+    _scalar = False
+    _linear = True
+    _fixed = False
+    _units = "(~)"
+    _print_value_fmt = "Shared parameters error: "
+
+    def __init__(
+        self,
+        things,
+        params=None,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        name="shared parameters",
+    ):
+        self._params = params
+        assert len(things) > 1, "only makes sense for >1 thing"
+        assert np.all([isinstance(things[0], type(t)) for t in things[1:]])
+        # TODO: some sort of check that all things are same resolution etc?
+        # something like _check_type for coilsets but more general...
+        # maybe can make a _check_type that takes in the user-inputted params.keys
+        # and just ensures that the params being fixed are the same resolution, that
+        # should be all we need to check I think
+        super().__init__(
+            things=things,
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            name=name,
+        )
+
+    def build(self, use_jit=False, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        thing = self.things[0]
+
+        # default params
+        default_params = tree_map(lambda dim: np.arange(dim), thing.dimensions)
+        self._params = setdefault(self._params, default_params)
+        self._params = broadcast_tree(self._params, default_params)
+        self._indices = tree_leaves(self._params)
+        assert tree_structure(self._params) == tree_structure(default_params)
+
+        self._dim_f = sum(idx.size for idx in self._indices) * (len(self.things) - 1)
+
+        # default target
+        self.target = np.zeros(self._dim_f)
```

Comment:
should I make sure that target does not get passed in ever? or is not including it in docstring enough to say that one should never pass it in? I worry someone will put a target of like what they wanna fix to and that will mess the objective up, as it would not really be caught anywhere (if the target is say a single number, as it would be broadcastable to the dim_f size)